fails to build: NewNet/nnsocket.h:89:22: error: ‘struct event’ has no member named ‘ev_flags’

system: gentoo 64bit
configure commandline:
~/museek-plus_/build $ cmake .. -DEVERYTHING=1

build log:
~/museek-plus_/build $ make
Scanning dependencies of target NewNet
[  0%] Building CXX object NewNet/CMakeFiles/NewNet.dir/nnbuffer.cpp.o
[  1%] Building CXX object NewNet/CMakeFiles/NewNet.dir/nnclientsocket.cpp.o
In file included from /home/andrius/museek-plus_/NewNet/nnclientsocket.h:24:0,
                 from /home/andrius/museek-plus_/NewNet/nnclientsocket.cpp:21:
/home/andrius/museek-plus_/NewNet/nnsocket.h: In constructor ‘NewNet::Socket::Socket()’:
/home/andrius/museek-plus_/NewNet/nnsocket.h:89:22: error: ‘struct event’ has no member named ‘ev_flags’
         m_EventData->ev_flags = 0; // This event has not been initialized
                      ^
make[2]: *** [NewNet/CMakeFiles/NewNet.dir/build.make:87: NewNet/CMakeFiles/NewNet.dir/nnclientsocket.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:86: NewNet/CMakeFiles/NewNet.dir/all] Error 2
make: *** [Makefile:128: all] Error 2

libevent version: 2.1.8
probably in new version API changed
I fixed it myself replacing `m_EventData->ev_flags = 0` with `memset(m_EventData, 0, sizeof(struct event))`.
it doesn't change meaning and works regardless if ev_flags exists or not.
will include pull request.

hey @andrius4669, i am having the same problem, so i replaced the line as you suggested but the error now becomes :
```
/home/k/bin/git/museek-plus/NewNet/nnsocket.h:90:3: error: ‘memset’ was not declared in this scope
   memset(m_EventData, 0, sizeof(struct event))
   ^~~~~~
/home/k/bin/git/museek-plus/NewNet/nnsocket.h:90:3: note: suggested alternative: ‘tzset’
   memset(m_EventData, 0, sizeof(struct event))
   ^~~~~~
   tzset
make[2]: *** [NewNet/CMakeFiles/NewNet.dir/build.make:87: NewNet/CMakeFiles/NewNet.dir/nnclientsocket.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:86: NewNet/CMakeFiles/NewNet.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```
Would you be so kind to point out what's wrong ?

--- a/NewNet/nnsocket.h
+++ b/NewNet/nnsocket.h
@@ -26,6 +26,7 @@
 #include "nnweakrefptr.h"
 #include "nnratelimiter.h"
 #include <event.h>
+#include <string.h>
 
 namespace NewNet
 {
@@ -86,7 +87,7 @@ namespace NewNet
               m_DataWaiting(false)
     {
         m_EventData = new struct event;
-        m_EventData->ev_flags = 0; // This event has not been initialized
+        memset(m_EventData, 0, sizeof(struct event));
     }
 
     //! Return the associated reactor.

